airflow/contrib/hooks/dingding_hook.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+
+import requests
+
+from airflow import AirflowException
+from airflow.hooks.http_hook import HttpHook
+
+
+class DingdingHook(HttpHook):
+    """
+    This hook allows you send Dingding message using Dingding custom bot.
+    Get Dingding token from conn_id.password. And prefer set domain to
+    conn_id.host, if not will use default ``https://oapi.dingtalk.com``.
+
+    For more detail message in
+    `Dingding custom bot <https://open-doc.dingtalk.com/microapp/serverapi2/qf2nxq>`_
+
+    :param dingding_conn_id: The name of the Dingding connection to use
+    :type dingding_conn_id: str
+    :param message_type: Message type you want to send to Dingding, support five type so far
+                         including text, link, markdown, actionCard, feedCard
+    :type message_type: str
+    :param message: The message send to Dingding chat group
+    :type message: str or dict
+    :param at_mobiles: Remind specific users with this message
+    :type at_mobiles: list of str
```

Comment:
`list[str]` is I think what we do else where. Can you check?

Reply:
I am sorry, I think you remind me last time due to I forget the `str` of `list of str`.
IMO, I think is same as `list of str` or `list[str]`? Am I wrong?

Reply:
Look at screenshot. 
`list[str]` is better, beacuse it works with [sphinx-internsphinx](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html)
![Screenshot 2019-03-25 at 12 36 57](https://user-images.githubusercontent.com/12058428/54916930-e6a6ea80-4efa-11e9-813c-f877dde2d0fe.png)

Reply:
@mik-laj Thanks your clarification, That help me a lot :+1: